lib/jsonapi/matchers/attributes_included.rb

```diff
@@ -0,0 +1,61 @@
+module Jsonapi
+  module Matchers
+    class AttributesIncluded
+      include Jsonapi::Matchers::Shared
+
+      def initialize(attribute_name, location)
+        @attribute_name = attribute_name
+        @location = location
+        @failure_message = nil
+      end
+
+      def with_value(expected_value)
+        @expected_value = expected_value
+        self
+      end
+
+      def matches?(target)
+        @target = normalize_target(target)
+        return false unless @target
+
+        if @location
+          @target = target.try(:[], :data).try(:[], @location).with_indifferent_access
+        else
+          @target = target.try(:[], :data).with_indifferent_access
+        end
+
+        @value = @target.try(:[], @attribute_name)
+
+
+        if @expected_value
+          if @expected_value.to_s == @value.to_s
+            return true
+          else
+            @failure_message = "expected '#{@expected_value}' for key '#{@attribute_name}', but got '#{@value}'"
+            return false
+          end
+        else
+          !!@value
+        end
+      end
+
+      def failure_message
+        @failure_message || "expected attribute '#{@attribute_name}' to be included in #{@target.as_json.ai}"
```

Comment:
glad you asked! its for pretty printing the hash. comes from awesomeprint

![screen shot 2016-08-23 at 12 05 51 pm](https://cloud.githubusercontent.com/assets/5402488/17901657/f4387902-6929-11e6-8249-4a5d2061ea2d.png)


 we also do this in the RecordIncluded matcher